Conformance tests are timing out waiting for Revision to be ready

## Expected Behavior

Conformance tests pass.

## Actual Behavior

Conformance tests timeout at step `The Revision will be updated when it is ready to serve traffic`.

## Steps to Reproduce the Problem

  1. Bring up an elafros cluster.
  2. Run the conformance tests.

## Additional Info

Elafros is working fine, to the point that a hello world app can be deployed and serves content.

Sample error output:

```
$ curl --header "Host:$SERVICE_HOST" http://${SERVICE_IP}
Hello World: shiniestnewestversion!
$ go test -v ./test/conformance -ginkgo.v
=== RUN   TestConformance
Running Suite: Conformance Suite
================================
Random Seed: 1522393799
Will run 1 of 1 specs

Route Deploying an app with a pre-built container 
  Creates a route, serves traffic to it, and serves traffic to subsequent revisions
  /go/src/github.com/elafros/elafros/test/conformance/route_test.go:190
STEP: Creating a new Route
STEP: Creating Configuration for the Route using the first image
STEP: The Configuration will be updated with the Revision after it is created
STEP: The Revision will be updated when it is ready to serve traffic

• Failure [120.715 seconds]
Route
/go/src/github.com/elafros/elafros/test/conformance/route_test.go:152
  Deploying an app with a pre-built container
  /go/src/github.com/elafros/elafros/test/conformance/route_test.go:189
    Creates a route, serves traffic to it, and serves traffic to subsequent revisions [It]
    /go/src/github.com/elafros/elafros/test/conformance/route_test.go:190

    Expected error:
        <*errors.errorString | 0xc4200cb6b0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    not to have occurred

    /go/src/github.com/elafros/elafros/test/conformance/crd_polling.go:78
------------------------------


Summarizing 1 Failure:

[Fail] Route Deploying an app with a pre-built container [It] Creates a route, serves traffic to it, and serves traffic to subsequent revisions 
/go/src/github.com/elafros/elafros/test/conformance/crd_polling.go:78

Ran 1 of 1 Specs in 120.864 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped --- FAIL: TestConformance (120.86s)
FAIL
FAIL    github.com/elafros/elafros/test/conformance     121.062s
```

--- a/pkg/controller/revision/BUILD.bazel
+++ b/pkg/controller/revision/BUILD.bazel
@@ -51,6 +51,7 @@ go_test(
     embed = [":go_default_library"],
     deps = [
         "//pkg/apis/build/v1alpha1:go_default_library",
+        "//pkg/apis/ela:go_default_library",
         "//pkg/apis/ela/v1alpha1:go_default_library",
         "//pkg/client/clientset/versioned/fake:go_default_library",
         "//pkg/client/informers/externalversions:go_default_library",

--- a/pkg/controller/revision/ela_service.go
+++ b/pkg/controller/revision/ela_service.go
@@ -17,6 +17,7 @@ limitations under the License.
 package revision
 
 import (
+	"github.com/elafros/elafros/pkg/apis/ela"
 	"github.com/elafros/elafros/pkg/apis/ela/v1alpha1"
 	"github.com/elafros/elafros/pkg/controller"
 
@@ -29,7 +30,7 @@ var httpServicePortName = "http"
 var servicePort = 80
 
 // MakeRevisionK8sService creates a Service that targets all pods with the same
-// "revision" label. Traffic is routed to queue-proxy port.
+// ela.RevisionLabelKey label. Traffic is routed to queue-proxy port.
 func MakeRevisionK8sService(u *v1alpha1.Revision, ns string) *corev1.Service {
 	return &corev1.Service{
 		ObjectMeta: meta_v1.ObjectMeta{
@@ -47,7 +48,7 @@ func MakeRevisionK8sService(u *v1alpha1.Revision, ns string) *corev1.Service {
 			},
 			Type: "NodePort",
 			Selector: map[string]string{
-				"revision": u.Name,
+				ela.RevisionLabelKey: u.Name,
 			},
 		},
 	}

--- a/pkg/controller/revision/revision.go
+++ b/pkg/controller/revision/revision.go
@@ -22,6 +22,8 @@ import (
 	"log"
 	"time"
 
+	"github.com/elafros/elafros/pkg/apis/ela"
+
 	"github.com/golang/glog"
 	"github.com/prometheus/client_golang/prometheus"
 	corev1 "k8s.io/api/core/v1"
@@ -880,8 +882,8 @@ func (c *Controller) updateStatus(rev *v1alpha1.Revision) (*v1alpha1.Revision, e
 // TODO: Consider using OwnerReferences.
 // https://github.com/kubernetes/sample-controller/blob/master/controller.go#L373-L384
 func lookupServiceOwner(endpoint *corev1.Endpoints) string {
-	// see if there's a 'revision' label on this object marking it as ours.
-	if revisionName, ok := endpoint.Labels["revision"]; ok {
+	// see if there's a label on this object marking it as ours.
+	if revisionName, ok := endpoint.Labels[ela.RevisionLabelKey]; ok {
 		return revisionName
 	}
 	return ""

--- a/pkg/controller/revision/revision_test.go
+++ b/pkg/controller/revision/revision_test.go
@@ -93,7 +93,7 @@ func getTestReadyEndpoints(revName string) *corev1.Endpoints {
 			Name:      fmt.Sprintf("%s-service", revName),
 			Namespace: testNamespace,
 			Labels: map[string]string{
-				"revision": revName,
+				ela.RevisionLabelKey: revName,
 			},
 		},
 		Subsets: []corev1.EndpointSubset{
@@ -114,7 +114,7 @@ func getTestAuxiliaryReadyEndpoints(revName string) *corev1.Endpoints {
 			Name:      fmt.Sprintf("%s-auxiliary", revName),
 			Namespace: "test",
 			Labels: map[string]string{
-				"revision": revName,
+				ela.RevisionLabelKey: revName,
 			},
 		},
 		Subsets: []corev1.EndpointSubset{
@@ -135,7 +135,7 @@ func getTestNotReadyEndpoints(revName string) *corev1.Endpoints {
 			Name:      fmt.Sprintf("%s-service", revName),
 			Namespace: "test",
 			Labels: map[string]string{
-				"revision": revName,
+				ela.RevisionLabelKey: revName,
 			},
 		},
 		Subsets: []corev1.EndpointSubset{

